### DISCUSSION TITLE
Support for Colpali

### DISCUSSION THREAD

**Discussion by amit-jain** [michaelfeil/infinity#389] [Ideas]
Is support for [Colpali]( https://github.com/illuin-tech/colpali ) been feasible? I chanced upon this repository for embedding inference and would be interested in Colpali and related adpater support. 
It probably needs additional endpoints though which may not align with the project.

**Comment by S1LV3RJ1NX** [comment]
the support is added I guess.

**Reply by michaelfeil** [reply]
Just added support! Still experimental in 0.0.68 - requires some testing with external libraries as comparison.

@S1LV3RJ1NX do you know any good libraries to unit-test against?

**Reply by simjak** [reply]
Is there a notebook on how I can use ColPali or ColQwen models?
Can I run those models in Runpod?
@michaelfeil

**Comment by simjak** [comment]
I tried to run  https://huggingface.co/vidore/colqwen2-v1.0 via RunPod, but got errors.

<img width="761" alt="image" src="https://github.com/user-attachments/assets/de5c517e-4d0e-45e3-a7d1-d9c078960694" />

**Reply by simjak** [reply]
here are more details:
https://github.com/michaelfeil/infinity/issues/507

**Reply by michaelfeil** [reply]
The version of Runpod is pinned to ~0.050 which does not allow multi model yet.

**Reply by simjak** [reply]
where can I update the version?

**Reply by michaelfeil** [reply]
https://github.com/runpod-workers/worker-infinity-embedding/blob/main/builder/requirements.txt

Be wary, infinity changed over time (python API), will take some effort.
